app/products/calls/connection/websocket_client.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {EventEmitter} from 'events';
+
+import {encode} from '@msgpack/msgpack/dist';
+
+import Calls from '@constants/calls';
+import DatabaseManager from '@database/manager';
+import {getCommonSystemValues} from '@queries/servers/system';
+
+export default class WebSocketClient extends EventEmitter {
+    private readonly serverUrl: string;
+    private readonly wsPath: string;
+    private ws: WebSocket | null = null;
+    private seqNo = 0;
+    private connID = '';
+    private eventPrefix = `custom_${Calls.PluginId}`;
+
+    constructor(serverUrl: string, wsPath: string) {
+        super();
+        this.serverUrl = serverUrl;
+        this.wsPath = wsPath;
+    }
+
+    async initialize() {
+        const database = DatabaseManager.serverDatabases[this.serverUrl]?.database;
+        if (!database) {
+            return;
+        }
+
+        const system = await getCommonSystemValues(database);
+        const connectionUrl = (system.config.WebsocketURL || this.serverUrl) + this.wsPath;
+
+        this.ws = new WebSocket(connectionUrl);
```

Comment:
5/5 We should be using `react-native-network-client` Websocket implementation here so all remote connections use the same library specially if this uses some sort of credentials / cookies / client certificates / etc...

Reply:
I have a rewritten `websocket_client.ts` that uses the `react-native-network-client` Websocket implementation, but there is a slight problem preventing me from committing it -- the  `react-native-network-client` only sends strings, and we're sending binary. We could convert, but that seems like it would be a performance concern. And it looks like we only need the bare native websocket at the moment (we're not using credentials / cookies / client certificates / etc... at the moment). (we will be adding auth for https://mattermost.atlassian.net/browse/MM-45750 , but that's one line).

Reply:
Ok, let's keep it as is for now as is working and auth is not needed at the moment, please do file a ticket for the mobile team to add support for sending binary data over the WS, so we can add it to the network library

Reply:
Done: https://mattermost.atlassian.net/browse/MM-46047